File: LeetCode/49.group-anagrams.rb
#
# @lc app=leetcode id=49 lang=ruby
#
# [49] Group Anagrams
#
# @param {String[]} strs
# @return {String[][]}
def group_anagrams(strs)
  anagrams = Hash.new { |hash, key| hash[key] = [] }
  strs.each do |str|
    sort_str = str.chars.sort
    anagrams[sort_str] << str
  end
  anagrams.values
end

File: LeetCode/49.group-anagrams.swift
/*
 * @lc app=leetcode id=49 lang=swift
 *
 * [49] Group Anagrams
 */

// @lc code=start
class Solution {
    func groupAnagrams(_ strs: [String]) -> [[String]] {
        return Array(Dictionary(grouping: strs, 
                                      by: { String($0.sorted())}).values)
    }
}
// @lc code=end

